Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+                    $FrontDoorResources += $frontDoorResource
+                }
+                catch
+                {
+                    Write-Host "Fetching Front Door resource: Resource Name: [$($resourceName)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                    Write-Host "Skipping this Front Door resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+                }
+            }
+        }
+    }
+    $totalFrontDoors = ($FrontDoorResources | Measure-Object).Count
+
+    if ($totalFrontDoors -eq 0)
+    {
+        Write-Host "No Front Door found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        return
+    }
+  
+    Write-Host "Found [$($totalFrontDoors)] Front Door(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetMinTLSVersionForCdnFrontDoors"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+    
+    # Includes required version of TLS
+    $requiredMinTLSVersion = "TLS12"
+
+    # Includes Front Doors where Minimum required TLS version is not set to 1.2.
+    $frontDoorsEndpointWithoutReqMinTLSVersion = @()
+
+    $frontDoorsEndpointWithReqMinTLSVersion = @()
+
+    # Includes Front Doors that were skipped during remediation. There were errors remediating them.
+    $frontDoorsSkipped = @()
+
+    $isMinTLSVersionSetOnCustomDomain = @()
+
+    $FrontDoorEndpoint = @()
+
+    $frontDoorWithoutReqMinTLSVersion = @()
+    $ResourceAppIdURI = "https://management.azure.com/"
+    $customDomainResponse =@()
+    $AccessMgtToken= (Get-AzAccessToken -ResourceUrl $ResourceAppIdURI).Token
+    
+    Write-Host "[Step 3 of 4] Fetch all Front Door configurations"
+    Write-Host $([Constants]::SingleDashLine)
+
+        $FrontDoorResources | ForEach-Object {
+                $frontDoorResource = $_
+                $resourceGroupName = $_.ResourceGroupName
+                $resourceName = $_.Name
+                $frontDoorEp = [FrontendEndPointsClass]:: new()
+                try
+                {
+                    Write-Host "Fetching Front Door Endpoint configuration: Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                
+                    $FrontDoorEndpoint = Get-AzFrontDoorCdnCustomDomain -ProfileName $resourceName -ResourceGroupName $resourceGroupName
```

Comment:
If $FrontDoorEndpoint is  a collection instead of single object, please renamed it to plural like $FrontDoorEndpoints

Reply:
Fixed